tests/test_callback.py

```diff
@@ -30,23 +34,54 @@ def run_playbook_callback(tmpdir):
     return run_playbook(playbook, inventory=inventory, extra_env=extra_env)
 
 
-def test_callback(tmpdir, vcrmode):
-    run = run_playbook_callback(tmpdir)
+def drop_incompatible_items(d):
+    """Recursively drop incompatible items before assertion"""
+    dd = {}
+    for k, v in d.items():
+        if isinstance(v, dict):
+            dd[k] = drop_incompatible_items(v)
+        elif isinstance(v, (list, set, tuple)):
+            dd[k] = type(v)(drop_incompatible_items(vv) if isinstance(vv, dict) else vv
+                            for vv in v)
+        elif re.match(r'^(?!msg|start|end|delta)', k):
+            dd[k] = v
+    return dd
+
+
+def run_callback(tmpdir, endpoint_type, vcrmode):
+    run = run_playbook_callback(tmpdir, endpoint_type)
     assert run.rc == 0
+    assert len(tmpdir.listdir()) > 0
     for real_file in tmpdir.listdir(sort=True):
         contents = real_file.read()
         contents = re.sub(r"\d+-\d+-\d+[ T]\d+:\d+:\d+\.\d+", "2000-01-01 12:00:00.0000", contents)
         contents = re.sub(r"\d+:\d+:\d+\.\d+", "12:00:00.0000", contents)
-        contents = re.sub(r", \\\"msg\\\": \\\"\\\"", "", contents)
+        if endpoint_type == "foreman":
+            # drop_incompatible_items cannot be used for the legacy format
```

Comment:
why not? you're doing it on line 77/78 (albeit on the JSON representation, not on the string one)

Reply:
Actually no, this is still needed. The `drop_incompatible_method` does not drill down to JSON-encoded strings :-)